lib/logstash/plugin_mixins/jdbc/scheduler.rb

```diff
@@ -0,0 +1,111 @@
+require 'rufus/scheduler'
+
+require 'logstash/util/loggable'
+
+module LogStash module PluginMixins module Jdbc
+  class Scheduler < Rufus::Scheduler
+
+    include LogStash::Util::Loggable
+
+    # Rufus::Scheduler >= 3.4 moved the Time impl into a gem EoTime = ::EtOrbi::EoTime`
+    # Rufus::Scheduler 3.1 - 3.3 using it's own Time impl `Rufus::Scheduler::ZoTime`
+    TimeImpl = defined?(Rufus::Scheduler::EoTime) ? Rufus::Scheduler::EoTime :
+                   (defined?(Rufus::Scheduler::ZoTime) ? Rufus::Scheduler::ZoTime : ::Time)
```

Comment:
for now LS is locked to rufus-scheduler `3.0.9` but this is compatible with any version `>= 3.0.9`
(there's an upgrade pending at: https://github.com/logstash-plugins/logstash-integration-jdbc/pull/97)